datamol/reactions/_attachments.py

```diff
@@ -0,0 +1,74 @@
+from typing import Union
+import re
+import datamol as dm
+import operator
+from rdkit import Chem
+
+ATTACHMENT_POINT_TOKEN = "*"
+ATTACHMENT_POINT_NUM_REGEXP = r"\[{}:?(\d*)\]".format(re.escape(ATTACHMENT_POINT_TOKEN))
+ATTACHMENT_POINT_REGEXP = r"(?:{0}|\[{0}[^\]]*\])".format(re.escape(ATTACHMENT_POINT_TOKEN))
+ATTACHMENT_POINT_NO_BRACKETS_REGEXP = r"(?<![:\[]){0}(?![:\]])".format(
+    re.escape(ATTACHMENT_POINT_TOKEN)
+)
+ALL_POSSIBLE_ATTACHMENTS = r"\[(\d*){}:?(\d*)\]".format(re.escape(ATTACHMENT_POINT_TOKEN))
+
+
+def add_brackets_to_attachment_points(smi: str):
```

Comment:
@zhu0619 , please add proper docs strings in the format used in datamol and better documentation overall. 


You might want to rebase since the PR seems unecessarily large.